examples/pipeline-evol-instruct-alpaca.py

```diff
@@ -41,35 +46,142 @@ def prepare(example):
+        The First Prompt: {first_instruction}
+        The Second Prompt: {second_instruction}
+        Your Judgement (Just answer: Equal or Not Equal. No need to explain the reason):"""
+
+    @dataclass
+    class EliminationEqualPrompts(TextGenerationTask):
+        """Task to check for the equality of two instructions following the Appendix G in
+        [WizardLM paper](https://arxiv.org/abs/2304.12244).
+        """
+
+        system_prompt: str = "You are an AI judge in charge of determining the equality of two instructions. "
+
+        def generate_prompt(self, input: List[str]) -> Prompt:
+            return Prompt(
+                system_prompt=self.system_prompt,
+                formatted_prompt=elimination_equal_prompt.format(
+                    first_instruction=input[0], second_instruction=input[1]
+                ),
+            )
+
+        def parse_output(self, output: str) -> List[Dict[str, str]]:
+            """Remove punctuation from the string and lowercase it."""
+            return {
+                "generations": output.translate(
+                    str.maketrans("", "", string.punctuation)
+                ).lower()
+            }
+
+    # Define our LLM for the elimination task
+    elimination_llm = OpenAILLM(
+        task=EliminationEqualPrompts(),
+        api_key=os.getenv("OPENAI_API_KEY"),
+        model="gpt-3.5-turbo",
+        num_threads=4,
+        max_new_tokens=2048,
     )
 
+    # Similarly to the paper, we will create a function to iterate M times over the dataset
+    # generating new instructions and then using the elimination task to remove the equal ones.
     start = time.time()
-    ds_evol_instruct_alpaca = pipe.generate(dataset, batch_size=8)
+
+    def make_evol_instruct_dataset(
+        evolver_llm: LLM,
+        elimination_llm: LLM,
+        dataset: Dataset,
+        evolution_steps: int = 4,
+    ) -> "Dataset":
+        # Set the pipelines
+        evolver_pipe = Pipeline(generator=evolver_llm)
+        elimination_pipe = Pipeline(generator=elimination_llm)
+
+        # Set the initial dataset
+        input_dataset = dataset
+        successful_instructions = []
+
+        # Start the evolution process
+        for step in range(1, evolution_steps + 1):
+            print(f"Evolving dataset step: {step}/{evolution_steps}")
+
+            # Generate new instructions
+            evolved_dataset = evolver_pipe.generate(input_dataset, batch_size=8)
+
+            # Prepare the dataset for the second pipeline with the original
+            # instructions and the evolved ones. If the evolved instruction is None,
+            # we use the original instruction (to make sure it will be removed)
+            def prepare_for_equal_prompts(example):
+                if example["instructions"][0] is None:
+                    return {"input": [example["input"], example["input"]]}
+                else:
+                    return {"input": [example["input"], example["instructions"][0]]}
```

Comment:
If the function is not changing I think we can put it outside the `for step in ...` loop.